packages/workspace/src/node/default-workspace-server.ts

```diff
@@ -46,32 +46,31 @@ export class WorkspaceCliContribution implements CliContribution {
 @injectable()
 export class DefaultWorkspaceServer implements WorkspaceServer {
 
-    protected root: Promise<string>;
+    protected root: Promise<string | undefined>;
 
     constructor(
         @inject(WorkspaceCliContribution) protected readonly cliParams: WorkspaceCliContribution,
         @inject(ILogger) protected readonly logger: ILogger
     ) {
-        this.root = this.getRootURI();
+        this.root = this.retrieveRootURI();
+    }
+
+    getRoot(): Promise<string | undefined> {
```

Comment:
Use a getter here?

Reply:
I don't think that getters are supported for JSON-RPC implementation

Reply:
ha right, forgot we were in a server for a second :)